docs/semgrep-secrets/conceptual-overview.md

```diff
@@ -0,0 +1,254 @@
+---
+slug: /semgrep-secrets/conceptual-overview
+append_help_link: true
+title: Conceptual overview
+hide_title: true
+description: Learn how Semgrep detects leaked secrets and helps you prioritize what keys to rotate.
+tags:
+  - Semgrep Secrets
+---
+
+import MoreHelp from "/src/components/MoreHelp"
+
+# Conceptual overview of Semgrep Secrets
+
+**Semgrep Secrets** is a code scanner that detects exposed API keys, passwords, and other credentials. These exposed secrets can be used by malicious actors to leak data or gain access to sensitive systems. Semgrep Secrets enables the user to know the following: 
+
+* What secrets have leaked.
+* The validation status of the secret, such as if it is confirmed valid. **Valid** secrets are secrets that are tested against a web service and confirmed to successfully grant resources or authentication. 
+* (For GitHub repositories) If these secrets are in public or private repositories.
+
+Save time and effort by prioritizing valid leaked secrets. Inform developers of valid, active secrets in their GitHub PRs by posting PR comments. 
+
+This document explains how Semgrep Secrets works and its approach to detecting secrets.
+
+:::info
+For a guide to setting up Semgrep Secrets, see [<i class="fa-regular fa-file-lines"></i> Getting started with Semgrep Secrets](/semgrep-secrets/getting-started).
+:::
+
+:::tip
+Secrets scanning is **language-agnostic**. This means Semgrep Secrets can scan any language for secrets.
+:::
+
+To ensure that findings are high-signal, comprehensive, and easy for users to prioritize, a secrets scan performs the following:
+
+* Semantic analysis
+* Validation
+* Entropy analysis
+
+The following sections explain how each analysis works.
+
+## Detecting secrets through semantic analysis
+
+Semantic analysis refers to Semgrep's ability to understand how data is used within your code. This differentiates Semgrep from regex-based detectors that simply define a pattern to match to a piece of code.
+
+Semgrep uses several mechanisms to perform semantic analysis. In particular, Semgrep uses **data-flow analysis**, which means that it is able to track data, such as variables, and the flow of that data across files and functions in your codebase. Semgrep is able to detect if a variable is renamed, unsanitized or sanitized, reassigned, or used in a function in such a way that a secret is exposed.
+
+See the following Semgrep rule and JavaScript test code for an example.
```

Comment:
I mistakenly committed this. I think we should keep "Semgrep rule" to make it clear that rules are what we use to perform semantic analysis. I also think (upon reflection) that we should still include at least 1 sentence of what a rule is, because some users may be coming to Semgrep for the first time for Secrets scanning.

Reply:
Agreed on the last point about the 1 sentence. See my other comment elsewhere for related discussion and overlap between Semgrep Secrets and Semgrep Code, and their rules.